src/types/multimin.rs

```diff
@@ -0,0 +1,297 @@
+n-dimensional functions. The algorithms proceed from an initial guess using a search algorithm which 
+attempts to move in a downhill direction.
+
+Algorithms making use of the gradient of the function perform a one-dimensional line minimisation along this 
+direction until the lowest point is found to a suitable tolerance. The search direction is then 
+updated with local information from the function and its derivatives, and the whole process repeated 
+until the true n-dimensional minimum is found.
+
+Algorithms which do not require the gradient of the function use different strategies. For example, 
+the Nelder-Mead Simplex algorithm maintains n+1 trial parameter vectors as the vertices of a n-dimensional 
+simplex. On each iteration it tries to improve the worst vertex of the simplex by geometrical transformations. 
+The iterations are continued until the overall size of the simplex has decreased sufficiently.
+
+Both types of algorithms use a standard framework. The user provides a high-level driver for the algorithms, 
+and the library provides the individual functions necessary for each of the steps. There are three 
+main phases of the iteration. The steps are,
+
+ * initialize minimizer state, s, for algorithm T
+ * update s using the iteration T
+ * test s for convergence, and repeat iteration if necessary
+
+Each iteration step consists either of an improvement to the line-minimisation in the current 
+direction or an update to the search direction itself. The state for the minimizers is held 
+in a gsl_multimin_fdfminimizer struct or a gsl_multimin_fminimizer struct.
+
+## Caveats
+
+Note that the minimization algorithms can only search for one local minimum at a time. When there are several 
+local minima in the search area, the first minimum to be found will be returned; however it is difficult 
+to predict which of the minima this will be. In most cases, no error will be reported if you try to 
+find a local minimum in an area where there is more than one.
+
+It is also important to note that the minimization algorithms find local minima; there is no way 
+to determine whether a minimum is a global minimum of the function in question.
+
+## Providing a function to minimize
+
+You must provide a parametric function of n variables for the minimizers to operate on. 
+
+*/
+
+use crate::{Value, VectorF64, View};
+use ffi::FFI;
+use sys;
+use sys::libc::c_void;
+
+ffi_wrapper!(
+    Minimizer<'a>,
+    *mut sys::gsl_multimin_fminimizer,
+    gsl_multimin_fminimizer_free
+    ;inner_call: sys::gsl_multimin_function_struct => sys::gsl_multimin_function_struct { f: None, n: 0_usize, params: std::ptr::null_mut() };
+    ;inner_closure: Option<Box<dyn Fn(&VectorF64) -> f64 + 'a>> => None;
+);
+
+impl<'a> Minimizer<'a> {
+    /// Creates a minimizer of type `t` for an n-dimensional function.
+	/// If there is insufficient memory to create the minimizer then 
```

Comment:
The indent of the doc comment is off (same below). Please fix it.

Reply:
Ok I replaced all \t with four-spaces. I am using vim.